buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -775,7 +775,7 @@ class BuildPlugin implements Plugin<Project> {
     static void applyCommonTestConfig(Project project) {
         project.tasks.withType(RandomizedTestingTask) {
             jvm "${project.runtimeJavaHome}/bin/java"
-            parallelism System.getProperty('tests.jvms', 'auto')
+            parallelism System.getProperty('tests.jvms',  String.valueOf(Runtime.getRuntime().availableProcessors() / 2))
```

Comment:
I have some concerns with this. Not all machines have hyper-threading, and not all machines have hyper-threading enabled. Some folks are choosing to disable hyper-threading because of security concerns. Intel is even releasing high-end processors that do not have hyper-threading (e.g., the i7-9700K, a workstation-class processor). On some machines, this will be too conservative.

Reply:
ES itself uses `Runtime.getRuntime().availableProcessors()` as the default for the number of processors. Is there any reason we shouldn't match it? Maybe because that'd be a bit aggressive for tests. Some of them are multi-threaded and they all benefit from multi-threaded GC.

Maybe we make a `hasHyperthreading` variable? It'd be OS dependant, but easy enough to check on linux and mac. I dunno about windows.

Reply:
Dropping my 2 cents in, I feel like we should go with `Runtime.getRuntime().availableProcessors()`. I think the optimal value varies based on hardware, but I found on an i7-6850k (6 core, 12 threads) that -Dtests.jvms=12 was faster than -Dtests.jvms=6; I'll caveat saying that the last time I tested this was 6 months ago and I didn't test values in between so there could be an optimal value between number of physical cores and number of threads that a processor can handle. Given my feeling that this is going to be hardware dependent, I think we could take the simplest approach first and if needed consider adding detection for hyper threading later on?

Reply:
CI has 8 cores 16 threads right now, and running with 8 is faster than with 16 for sure.
It might not be related to hardware alone, but also to the number of the tests, as if there are few tests per worker the cost of starting up those workers is not worth it. 

One can easily add `systemProp.tests.jvms=6` to `~/.gradle/gradle.properties` to customize this for their own hardware if the default is not appropriate.

@jasontedor I'm not entirely convinced it will be too conservative. On the processor you mentioned it will make no difference  as it has 8 cores, and we'll continue using 4 on that processor,  so there will be no change, as we currently cap at 4. 
We will use fewer test workers on CPUs with fewer than 8 cores that have no or disabled hyper-threading. 

I don't have a strong preference on the value we set this to. We could set it to the number of processors, but would prefer not to do hyper-threading detection ( or anything OS specific for that matter ) until we are absolutely sure we need it. 

Maybe adding some documentation in `TESTING` will be enough.

Reply:
@jasontedor do you think that we should implement hyper-threading detection now ? I was trying to make the point that while the configuration will not be optimal for users with hyper threading disabled or absent, it will still be the same or close as we have Today as we cap to 4 threads right now.  If you agree that this is the case we could start with this change to get some benefits in CI right away and then look at what else we can do to have a more generally useful default if this is common enough that we feel having  `systemProp.tests.jvms` documented  will not be sufficient.